components/CardDetail/CardDetail.js

```diff
@@ -1,10 +1,24 @@
 import styled from "styled-components";
 
-export default function CardDetails() {
+export default function CardDetails({ updateCardDetail, id, showMoreDetails }) {
   function handleDetailSubmit(event) {
     event.preventDefault();
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
+    updateCardDetail(
+      id,
+      data.result,
+      { Player1: data.firstsetplayerone, Player2: data.firstsetplayertwo },
+      {
+        Player1: data.secondsetplayerone,
+        Player2: data.secondsetplayertwo,
+      },
+      {
+        Player1: data.thirdsetplayerone,
+        Player2: data.thirdsetplayertwo,
+      }
+    );
+    showMoreDetails();
   }
```

Comment:
This is where you need to look into validating your data 👀 😄 

Check for input that is insensible and don't let it slip by :)